`make install` fails due to incompatible dependency configuration

[x] I have checked the [documentation](https://docs.ragas.io/) and related resources and couldn't resolve my bug.

---

## Description

The current development setup has an incompatibility between the root Makefile and `ragas/pyproject.toml` that prevents contributors from setting up the development environment as documented.

### Current Behavior

Running `make install` fails with:
```bash
$ make install
Installing dependencies...
Installing ragas dependencies (including experimental)...
...
warning: The package `ragas @ file:///home/max/ragas/ragas` does not have an extra named `dev`
Setting up pre-commit hooks...
make: pre-commit: No such file or directory
make: *** [Makefile:21: install] Error 127
```

### Environment
- OS: MX Linux 23
- Python: 3.11.2
- uv version: 0.8.11
- (RAGAS version /) Repository commit: f16676764 (main branch)

### Root Cause

1. **Dependency format mismatch**: The root Makefile attempts to install using extras syntax:
   ```makefile
   $(Q)uv pip install -e "./ragas[dev]"
   ```
   However, `ragas/pyproject.toml` defines dev dependencies using `[dependency-groups]` (PEP 735) instead of `[project.optional-dependencies]`, so there is no `dev` extra available. This means `pre-commit` is never installed, causing the hook setup to fail.

2. **Directory structure issue**: When using `uv pip install --group dev`, uv expects a `pyproject.toml` in the current working directory, but in the ragas repository structure, it's located at `ragas/pyproject.toml`. The root Makefile needs to either use `--directory ragas` or `cd ragas` to access it.

## Steps to Reproduce

1. Clone the repository
2. Install uv: `curl -LsSf https://astral.sh/uv/install.sh | sh` 
3. Create venv: `uv venv`
4. Activate: `source .venv/bin/activate`
5. Run `make install`
6. Observe the error

## Expected Behavior

`make install` should successfully install all necessary development dependencies and set up pre-commit hooks.

## Potential Solutions

### Option 1: Use dependency-groups with correct directory (leads to excessive downloads)
```makefile
$(Q)uv pip install -e "./ragas"
$(Q)cd ragas && uv pip install --group dev
# OR: $(Q)uv pip install --directory ragas --group dev
```

**Problem**: This installs 236+ packages including PyTorch, CUDA libraries (~1.5GB download) because the dev group includes:
```toml
"ragas[all,tracing,gdrive,ai-frameworks]"
```
This might be excessive for contributors who just want to run tests, format code, and submit PRs?

### Option 2: Convert to project.optional-dependencies
Move from `[dependency-groups]` to `[project.optional-dependencies]` in `ragas/pyproject.toml`:
```toml
[project.optional-dependencies]
dev = [
    "ruff",
    "pyright>=1.1.403",
    "pre-commit>=4.3.0",
    "pytest",
    "pytest-xdist[psutil]",
    "pytest-asyncio",
    "nbmake",
    "notebook",
    # NOT including "ragas[all,...]" to keep it minimal
]
```

Then the Makefile can use:
```makefile
$(Q)uv pip install -e "./ragas[dev]"
```

### Option 3: Separate minimal and full dev installations (Recommended)
Create two installation targets - one for basic development needs, one for full ML stack:

```makefile
install: ## Install minimal dev dependencies for contributing
	@echo "Installing ragas core..."
	$(Q)uv pip install -e "./ragas"
	@echo "Installing essential dev tools..."
	$(Q)uv pip install pre-commit ruff pyright pytest pytest-xdist pytest-asyncio nbmake
	@echo "Setting up pre-commit hooks..."
	$(Q)uv run pre-commit install
	@echo "Installation complete! (minimal dev environment)"

install-full: ## Install ALL dependencies including ML frameworks (~1.5GB)
	@echo "Installing ragas core..."
	$(Q)uv pip install -e "./ragas"
	@echo "Installing ALL dev dependencies (includes ML frameworks)..."
	$(Q)cd ragas && uv pip install --group dev
	@echo "Setting up pre-commit hooks..."
	$(Q)uv run pre-commit install
	@echo "Full installation complete!"
```

### Option 4: Hybrid with both dependency formats
Maintain both formats for different use cases:
```toml
[project.optional-dependencies]
dev = [  # Minimal dev requirements
    "ruff",
    "pyright>=1.1.403",
    "pre-commit>=4.3.0",
    "pytest",
    "pytest-xdist[psutil]",
    "pytest-asyncio",
    "nbmake",
]

[dependency-groups]
dev-full = [  # Everything including ML stack
    "ragas[all,tracing,gdrive,ai-frameworks]",
    # ... other heavy dependencies
]
```

## Questions

Which option would you prefer? Do all contributors immediatelly need full heavy ML stacks?




## Additional Context

The DEVELOPMENT.md instructs users to run `make install`, but this currently doesn't work out of the box. This is a barrier for new contributors wanting to help with the project. The fix should ensure a smooth onboarding experience while keeping the option for full installations when needed.

See #2164

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -108,7 +108,7 @@ jobs:
 
       - name: Install dependencies
         run: |
-          # Use minimal install for fast CI runs (49 packages vs 383)
+          # Use minimal install for fast CI runs (79 packages vs 383)
           # This uses make install-minimal for consistency with local development
           make install-minimal
 
@@ -171,7 +171,7 @@ jobs:
 
       - name: Install dependencies
         run: |
-          # Use minimal install for fast CI runs (49 packages vs 383)
+          # Use minimal install for fast CI runs (79 packages vs 383)
           # This uses make install-minimal for consistency with local development
           make install-minimal
 

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -108,8 +108,9 @@ jobs:
 
       - name: Install dependencies
         run: |
-          # Use UV with system installation for CI (simpler and more reliable)
-          uv pip install --system -e "." --group dev --cache-dir ~/.cache/uv
+          # Use minimal install for fast CI runs (83 packages vs 309)
+          # This uses make install-minimal for consistency with local development
+          make install-minimal
 
       - name: Run unit tests
         run: |
@@ -170,8 +171,9 @@ jobs:
 
       - name: Install dependencies
         run: |
-          # Use UV with system installation for CI (simpler and more reliable)
-          uv pip install --system -e "." --group dev --cache-dir ~/.cache/uv
+          # Use minimal install for fast CI runs (83 packages vs 309)
+          # This uses make install-minimal for consistency with local development
+          make install-minimal
 
       - name: Format check (dry run)
         run: |

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -123,10 +123,10 @@ jobs:
           # Run different test suites based on test type
           if [ "${{ matrix.test-type }}" = "full" ]; then
             # Full test suite with notebook tests
-            pytest --nbmake tests/unit tests/experimental "${OPTS[@]}"
+            uv run pytest --nbmake tests/unit tests/experimental "${OPTS[@]}"
           else
             # Essential tests only (faster for cross-platform validation)
-            pytest tests/unit tests/experimental -k "not slow" "${OPTS[@]}"
+            uv run pytest tests/unit tests/experimental -k "not slow" "${OPTS[@]}"
           fi
         env:
           __RAGAS_DEBUG_TRACKING: true
@@ -181,8 +181,8 @@ jobs:
           # Note: We use direct commands here instead of the standalone Makefiles
           # to have precise control over CI-specific options like --check for dry-run
           echo "Checking ragas formatting..."
-          ruff format --check src tests docs --exclude src/ragas/_version.py --config pyproject.toml
-          ruff check src docs tests --exclude src/ragas/_version.py --config pyproject.toml
+          uv run ruff format --check src tests docs --exclude src/ragas/_version.py --config pyproject.toml
+          uv run ruff check src docs tests --exclude src/ragas/_version.py --config pyproject.toml
 
       - name: Type check
         run: make type

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -108,7 +108,7 @@ jobs:
 
       - name: Install dependencies
         run: |
-          # Use minimal install for fast CI runs (83 packages vs 309)
+          # Use minimal install for fast CI runs (49 packages vs 383)
           # This uses make install-minimal for consistency with local development
           make install-minimal
 
@@ -171,7 +171,7 @@ jobs:
 
       - name: Install dependencies
         run: |
-          # Use minimal install for fast CI runs (83 packages vs 309)
+          # Use minimal install for fast CI runs (49 packages vs 383)
           # This uses make install-minimal for consistency with local development
           make install-minimal
 

--- a/CLAUDE.md
+++ b/CLAUDE.md
@@ -19,7 +19,7 @@ The repository contains:
 Choose the appropriate installation based on your needs:
 
 ```bash
-# RECOMMENDED: Minimal dev setup (49 packages - fast)
+# RECOMMENDED: Minimal dev setup (79 packages - fast)
 make install-minimal
 
 # FULL: Complete dev environment (383 packages - comprehensive)  
@@ -49,7 +49,7 @@ uv sync --group dev
 
 ```bash
 # Setup and installation  
-make install-minimal # Minimal dev setup (49 packages - recommended)
+make install-minimal # Minimal dev setup (79 packages - recommended)
 make install        # Full dev environment (383 packages - complete)
 
 # Code quality
@@ -194,6 +194,6 @@ analytics_logger.addHandler(console_handler)
 - whenever you create such docs put in in /\_experiments because that is gitignored and you can use it as a scratchpad or tmp directory for storing these
 - always use uv to run python and python related commandline tools like isort, ruff, pyright etc. This is because we are using uv to manage the .venv and dependencies.
 - The project uses two distinct dependency management approaches:
-  - **Minimal setup**: `[project.optional-dependencies].dev-minimal` for fast development (49 packages)
+  - **Minimal setup**: `[project.optional-dependencies].dev-minimal` for fast development (79 packages)
   - **Full setup**: `[dependency-groups].dev` for comprehensive development (383 packages)
 - Use `make install-minimal` for most development tasks, `make install` for full ML stack work

--- a/CLAUDE.md
+++ b/CLAUDE.md
@@ -16,23 +16,41 @@ The repository contains:
 
 ### Installation
 
+Choose the appropriate installation based on your needs:
+
 ```bash
+# RECOMMENDED: Minimal dev setup (49 packages - fast)
+make install-minimal
+
+# FULL: Complete dev environment (383 packages - comprehensive)  
+make install
+
+# OR manual installation:
 # Create a virtual environment
 python -m venv venv
 source venv/bin/activate  # On Windows, use `venv\Scripts\activate`
 
-# For ragas (includes experimental features)
-pip install -U setuptools  # Required on newer Python versions
-pip install -e ".[dev]"
+# Minimal dev setup (uses [project.optional-dependencies].dev-minimal)
+uv pip install -e ".[dev-minimal]"
+
+# Full dev setup (uses [dependency-groups].dev)
+uv sync --group dev
 ```
+
+### Installation Methods Explained
+
+- **Minimal setup**: Uses `uv pip install` with optional dependencies for selective installation
+- **Full setup**: Uses `uv sync` with dependency groups for comprehensive environment management
+- **No naming conflicts**: `dev-minimal` vs `dev` clearly distinguish the two approaches
 
 ## Common Commands
 
 ### Commands (from root directory)
 
 ```bash
-# Setup and installation
-make install        # Install dependencies for ragas
+# Setup and installation  
+make install-minimal # Minimal dev setup (49 packages - recommended)
+make install        # Full dev environment (383 packages - complete)
 
 # Code quality
 make format         # Format and lint all code
@@ -174,4 +192,8 @@ analytics_logger.addHandler(console_handler)
 ## Memories
 
 - whenever you create such docs put in in /\_experiments because that is gitignored and you can use it as a scratchpad or tmp directory for storing these
-- always use uv to run python and python related commandline tools like isort, ruff, pyright ect. This is because we are using uv to manage the .venv and dependencies.
+- always use uv to run python and python related commandline tools like isort, ruff, pyright etc. This is because we are using uv to manage the .venv and dependencies.
+- The project uses two distinct dependency management approaches:
+  - **Minimal setup**: `[project.optional-dependencies].dev-minimal` for fast development (49 packages)
+  - **Full setup**: `[dependency-groups].dev` for comprehensive development (383 packages)
+- Use `make install-minimal` for most development tasks, `make install` for full ML stack work

--- a/DEVELOPMENT.md
+++ b/DEVELOPMENT.md
@@ -14,10 +14,10 @@ curl -LsSf https://astral.sh/uv/install.sh | sh
 
 # 3. Choose your installation type:
 
-# RECOMMENDED: Minimal dev setup (83 packages - fast)  
+# RECOMMENDED: Minimal dev setup (49 packages - fast)  
 make install-minimal
 
-# FULL: Complete dev environment (309+ packages - comprehensive)
+# FULL: Complete dev environment (383 packages - comprehensive)
 make install
 
 # 4. Verify everything works
@@ -34,8 +34,8 @@ AI agents working with this codebase should use these standardized commands:
 ```bash
 # Essential commands for AI development
 make help           # See all available targets
-make install-minimal # Minimal dev setup (fast - 83 packages) 
-make install        # Full environment (modern uv sync - 309+ packages)
+make install-minimal # Minimal dev setup (fast - 49 packages) 
+make install        # Full environment (modern uv sync - 383 packages)
 make check          # Quick health check (format + type)
 make test           # Run all tests
 make run-ci         # Full CI pipeline locally
@@ -86,10 +86,10 @@ This repository is organized as a single project with integrated experimental fe
 
 #### Option 1: Using Make (Recommended)
 ```bash
-# Recommended: Minimal dev setup (83 packages)  
+# Recommended: Minimal dev setup (49 packages)  
 make install-minimal
 
-# Full: Complete environment (309+ packages)
+# Full: Complete environment (383 packages)
 make install
 ```
 
@@ -99,7 +99,7 @@ make install
 curl -LsSf https://astral.sh/uv/install.sh | sh
 
 # Minimal dev: Core + essential dev tools
-uv pip install -e ".[dev]"
+uv pip install -e ".[dev-minimal]"
 
 # Full dev: Everything (uses modern uv sync)
 uv sync --group dev
@@ -107,13 +107,13 @@ uv sync --group dev
 
 #### Which Option to Choose?
 
-**Use `make install-minimal` (83 packages) if you're:**
+**Use `make install-minimal` (49 packages) if you're:**
 - Contributing to ragas development
 - Need testing and linting tools
 - Want fast CI/CD builds
 - Working on code quality, docs, or basic features
 
-**Use `make install` (309+ packages) if you're:**
+**Use `make install` (383 packages) if you're:**
 - Working on ML features requiring the full stack
 - Need observability tools (Phoenix, MLflow)
 - Developing with notebooks and advanced integrations
@@ -121,8 +121,8 @@ uv sync --group dev
 
 #### Installation Methods Explained
 
-- **`install-minimal`**: Uses `uv pip install` for selective dev dependencies  
-- **`install`**: Uses `uv sync` for complete modern dependency management
+- **`install-minimal`**: Uses `uv pip install -e ".[dev-minimal]"` for selective minimal dev dependencies (49 packages)
+- **`install`**: Uses `uv sync --group dev` for complete modern dependency management (383 packages)
 
 ### Verification
 ```bash
@@ -135,8 +135,8 @@ make test   # Runs all tests
 Run `make help` to see all targets. Here are the essential commands:
 
 ### Setup & Installation
-- `make install-minimal` - Install minimal dev setup (83 packages - recommended)
-- `make install` - Install full environment with uv sync (309+ packages - complete)
+- `make install-minimal` - Install minimal dev setup (49 packages - recommended)
+- `make install` - Install full environment with uv sync (383 packages - complete)
 
 ### Code Quality
 - `make format` - Format and lint all code (includes unused import cleanup)

--- a/DEVELOPMENT.md
+++ b/DEVELOPMENT.md
@@ -12,7 +12,12 @@ cd ragas
 # 2. Install uv (if not already installed)
 curl -LsSf https://astral.sh/uv/install.sh | sh
 
-# 3. Install dependencies and setup environment
+# 3. Choose your installation type:
+
+# RECOMMENDED: Minimal dev setup (83 packages - fast)  
+make install-minimal
+
+# FULL: Complete dev environment (309+ packages - comprehensive)
 make install
 
 # 4. Verify everything works
@@ -29,7 +34,8 @@ AI agents working with this codebase should use these standardized commands:
 ```bash
 # Essential commands for AI development
 make help           # See all available targets
-make install        # Install dependencies and setup environment
+make install-minimal # Minimal dev setup (fast - 83 packages) 
+make install        # Full environment (modern uv sync - 309+ packages)
 make check          # Quick health check (format + type)
 make test           # Run all tests
 make run-ci         # Full CI pipeline locally
@@ -80,6 +86,10 @@ This repository is organized as a single project with integrated experimental fe
 
 #### Option 1: Using Make (Recommended)
 ```bash
+# Recommended: Minimal dev setup (83 packages)  
+make install-minimal
+
+# Full: Complete environment (309+ packages)
 make install
 ```
 
@@ -88,9 +98,31 @@ make install
 # Install uv if not available
 curl -LsSf https://astral.sh/uv/install.sh | sh
 
-# Install the project
+# Minimal dev: Core + essential dev tools
+uv pip install -e ".[dev]"
+
+# Full dev: Everything (uses modern uv sync)
 uv sync --group dev
 ```
+
+#### Which Option to Choose?
+
+**Use `make install-minimal` (83 packages) if you're:**
+- Contributing to ragas development
+- Need testing and linting tools
+- Want fast CI/CD builds
+- Working on code quality, docs, or basic features
+
+**Use `make install` (309+ packages) if you're:**
+- Working on ML features requiring the full stack
+- Need observability tools (Phoenix, MLflow)
+- Developing with notebooks and advanced integrations
+- Want the complete development environment
+
+#### Installation Methods Explained
+
+- **`install-minimal`**: Uses `uv pip install` for selective dev dependencies  
+- **`install`**: Uses `uv sync` for complete modern dependency management
 
 ### Verification
 ```bash
@@ -103,7 +135,8 @@ make test   # Runs all tests
 Run `make help` to see all targets. Here are the essential commands:
 
 ### Setup & Installation
-- `make install` - Install dependencies for both projects
+- `make install-minimal` - Install minimal dev setup (83 packages - recommended)
+- `make install` - Install full environment with uv sync (309+ packages - complete)
 
 ### Code Quality
 - `make format` - Format and lint all code (includes unused import cleanup)

--- a/DEVELOPMENT.md
+++ b/DEVELOPMENT.md
@@ -14,7 +14,7 @@ curl -LsSf https://astral.sh/uv/install.sh | sh
 
 # 3. Choose your installation type:
 
-# RECOMMENDED: Minimal dev setup (49 packages - fast)  
+# RECOMMENDED: Minimal dev setup (79 packages - fast)  
 make install-minimal
 
 # FULL: Complete dev environment (383 packages - comprehensive)
@@ -34,7 +34,7 @@ AI agents working with this codebase should use these standardized commands:
 ```bash
 # Essential commands for AI development
 make help           # See all available targets
-make install-minimal # Minimal dev setup (fast - 49 packages) 
+make install-minimal # Minimal dev setup (fast - 79 packages) 
 make install        # Full environment (modern uv sync - 383 packages)
 make check          # Quick health check (format + type)
 make test           # Run all tests
@@ -86,7 +86,7 @@ This repository is organized as a single project with integrated experimental fe
 
 #### Option 1: Using Make (Recommended)
 ```bash
-# Recommended: Minimal dev setup (49 packages)  
+# Recommended: Minimal dev setup (79 packages)  
 make install-minimal
 
 # Full: Complete environment (383 packages)
@@ -107,7 +107,7 @@ uv sync --group dev
 
 #### Which Option to Choose?
 
-**Use `make install-minimal` (49 packages) if you're:**
+**Use `make install-minimal` (79 packages) if you're:**
 - Contributing to ragas development
 - Need testing and linting tools
 - Want fast CI/CD builds
@@ -121,7 +121,7 @@ uv sync --group dev
 
 #### Installation Methods Explained
 
-- **`install-minimal`**: Uses `uv pip install -e ".[dev-minimal]"` for selective minimal dev dependencies (49 packages)
+- **`install-minimal`**: Uses `uv pip install -e ".[dev-minimal]"` for selective minimal dev dependencies (79 packages)
 - **`install`**: Uses `uv sync --group dev` for complete modern dependency management (383 packages)
 
 ### Verification
@@ -135,7 +135,7 @@ make test   # Runs all tests
 Run `make help` to see all targets. Here are the essential commands:
 
 ### Setup & Installation
-- `make install-minimal` - Install minimal dev setup (49 packages - recommended)
+- `make install-minimal` - Install minimal dev setup (79 packages - recommended)
 - `make install` - Install full environment with uv sync (383 packages - complete)
 
 ### Code Quality

--- a/Makefile
+++ b/Makefile
@@ -19,17 +19,17 @@ setup-venv: ## Set up uv virtual environment
 	@echo "Virtual environment created at .venv"
 	@echo "To activate: source .venv/bin/activate"
 
-install-minimal: ## Install minimal dev dependencies (fast setup - 83 packages)
+install-minimal: ## Install minimal dev dependencies (fast setup - 49 packages)
 	@echo "Installing minimal development dependencies (fast setup)..."
 	@if [ ! -d ".venv" ]; then \
 		echo "Virtual environment not found, creating one..."; \
 		$(MAKE) setup-venv; \
 	fi
 	@echo "Installing core ragas + essential dev tools..."
-	$(Q)uv pip install -e ".[dev]"
+	$(Q)uv pip install -e ".[dev-minimal]"
 	@echo "Setting up pre-commit hooks..."
 	$(Q)uv run pre-commit install
-	@echo "Minimal installation complete! (83 packages)"
+	@echo "Minimal installation complete! (49 packages)"
 	@echo "Note: For full features including ML packages, use 'make install'"
 
 install: ## Install full dependencies with uv sync (backward compatible - modern approach)

--- a/Makefile
+++ b/Makefile
@@ -19,23 +19,36 @@ setup-venv: ## Set up uv virtual environment
 	@echo "Virtual environment created at .venv"
 	@echo "To activate: source .venv/bin/activate"
 
-install: ## Install dependencies for ragas
-	@echo "Installing dependencies..."
+install-minimal: ## Install minimal dev dependencies (fast setup - 83 packages)
+	@echo "Installing minimal development dependencies (fast setup)..."
 	@if [ ! -d ".venv" ]; then \
 		echo "Virtual environment not found, creating one..."; \
 		$(MAKE) setup-venv; \
 	fi
-	@echo "Installing ragas dependencies..."
+	@echo "Installing core ragas + essential dev tools..."
+	$(Q)uv pip install -e ".[dev]"
+	@echo "Setting up pre-commit hooks..."
+	$(Q)uv run pre-commit install
+	@echo "Minimal installation complete! (83 packages)"
+	@echo "Note: For full features including ML packages, use 'make install'"
+
+install: ## Install full dependencies with uv sync (backward compatible - modern approach)
+	@echo "Installing full development dependencies with uv sync..."
+	@if [ ! -d ".venv" ]; then \
+		echo "Virtual environment not found, creating one..."; \
+		$(MAKE) setup-venv; \
+	fi
+	@echo "Installing ragas with full dev environment..."
 	$(Q)VIRTUAL_ENV= uv sync --group dev
 	@echo "Setting up pre-commit hooks..."
-	$(Q)uv run --active pre-commit install
-	@echo "Installation complete!"
+	$(Q)uv run pre-commit install
+	@echo "Full installation complete! (Modern uv sync approach)"
 
 # =============================================================================
 # CODE QUALITY
 # =============================================================================
 
-.PHONY: help setup-venv install format type check clean test test-e2e benchmarks benchmarks-docker run-ci run-ci-fast run-ci-format-check run-ci-type run-ci-tests build-docs serve-docs process-experimental-notebooks
+.PHONY: help setup-venv install-minimal install format type check clean test test-e2e benchmarks benchmarks-docker run-ci run-ci-fast run-ci-format-check run-ci-type run-ci-tests build-docs serve-docs process-experimental-notebooks
 format: ## Format and lint all code
 	@echo "Formatting and linting all code..."
 	@echo "(ruff format) Formatting ragas..."

--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@ setup-venv: ## Set up uv virtual environment
 	@echo "Virtual environment created at .venv"
 	@echo "To activate: source .venv/bin/activate"
 
-install-minimal: ## Install minimal dev dependencies (fast setup - 49 packages)
+install-minimal: ## Install minimal dev dependencies (fast setup - 79 packages)
 	@echo "Installing minimal development dependencies (fast setup)..."
 	@if [ ! -d ".venv" ]; then \
 		echo "Virtual environment not found, creating one..."; \
@@ -29,7 +29,7 @@ install-minimal: ## Install minimal dev dependencies (fast setup - 49 packages)
 	$(Q)uv pip install -e ".[dev-minimal]"
 	@echo "Setting up pre-commit hooks..."
 	$(Q)uv run pre-commit install
-	@echo "Minimal installation complete! (49 packages)"
+	@echo "Minimal installation complete! (79 packages)"
 	@echo "Note: For full features including ML packages, use 'make install'"
 
 install: ## Install full dependencies with uv sync (backward compatible - modern approach)

--- a/examples/benchmark_llm/evals.py
+++ b/examples/benchmark_llm/evals.py
@@ -7,12 +7,12 @@ import sys
 from typing import List, Optional
 
 import pandas as pd
-from ragas.experimental import Dataset
-from ragas import experiment
-from ragas.experimental.metrics.result import MetricResult
-from ragas.experimental.metrics.discrete import discrete_metric
+from prompt import DEFAULT_MODEL, run_prompt
 
-from prompt import run_prompt, DEFAULT_MODEL
+from ragas import experiment
+from ragas.experimental import Dataset
+from ragas.experimental.metrics.discrete import discrete_metric
+from ragas.experimental.metrics.result import MetricResult
 
 
 @discrete_metric(name="discount_accuracy", allowed_values=["correct", "incorrect"])

--- a/examples/gdrive_append_example.py
+++ b/examples/gdrive_append_example.py
@@ -5,6 +5,7 @@ while preserving the existing records.
 """
 
 from pydantic import BaseModel
+
 from ragas.dataset import Dataset
 
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -73,6 +73,7 @@ dev-minimal = [
     "pytest",
     "pytest-xdist[psutil]",
     "pytest-asyncio",
+    "nbmake",
     "build>=1.3.0",
 ]
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -65,15 +65,15 @@ gdrive = [
 ]
 ai-frameworks = ["haystack-ai"]
 
-# Minimal dev dependencies for fast development setup
-dev = [
+# Minimal dev dependencies for fast development setup (used by make install-minimal)
+dev-minimal = [
     "ruff",
     "pyright>=1.1.403",
     "pre-commit>=4.3.0",
     "pytest",
     "pytest-xdist[psutil]",
     "pytest-asyncio",
-    "nbmake",
+    "build>=1.3.0",
 ]
 
 test = []
@@ -145,31 +145,23 @@ markers = [
 ]
 
 [dependency-groups]
-# Minimal dev dependencies for fast setup (essential tools only)
-dev-minimal = [
-    "ruff",
-    "pyright>=1.1.403", 
-    "pre-commit>=4.3.0",
-    "pytest",
-    "pytest-xdist[psutil]",
-    "pytest-asyncio",
-    "build>=1.3.0",
-]
-
-# Full dev dependencies with all features (backward compatible)
+# Full dev dependencies with all features (used by make install)
 dev = [
+    # Core dev tools (shared with minimal)
     "ruff",
     "pyright>=1.1.403",
-    "pre-commit>=4.3.0",
+    "pre-commit>=4.3.0", 
     "pytest",
     "pytest-xdist[psutil]",
     "pytest-asyncio",
+    "build>=1.3.0",
+    # Additional tools for full dev
     "nbmake",
-    "notebook",
+    "notebook", 
     "arize-phoenix>=6.1.0",
     "openinference-instrumentation-langchain>=0.1.29",
+    # Include all optional features
     "ragas[all,tracing,gdrive,ai-frameworks]",
-    "build>=1.3.0",
 ]
 docs = [
     "mkdocs>=1.6.1",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -65,6 +65,17 @@ gdrive = [
 ]
 ai-frameworks = ["haystack-ai"]
 
+# Minimal dev dependencies for fast development setup
+dev = [
+    "ruff",
+    "pyright>=1.1.403",
+    "pre-commit>=4.3.0",
+    "pytest",
+    "pytest-xdist[psutil]",
+    "pytest-asyncio",
+    "nbmake",
+]
+
 test = []
 [project.entry-points."ragas.backends"]
 "local/csv" = "ragas.backends.local_csv:LocalCSVBackend"
@@ -134,6 +145,18 @@ markers = [
 ]
 
 [dependency-groups]
+# Minimal dev dependencies for fast setup (essential tools only)
+dev-minimal = [
+    "ruff",
+    "pyright>=1.1.403", 
+    "pre-commit>=4.3.0",
+    "pytest",
+    "pytest-xdist[psutil]",
+    "pytest-asyncio",
+    "build>=1.3.0",
+]
+
+# Full dev dependencies with all features (backward compatible)
 dev = [
     "ruff",
     "pyright>=1.1.403",

